pkg/scheduler/internal/cache/interface.go

```diff
@@ -19,8 +19,8 @@ package cache
 import (
 	v1 "k8s.io/api/core/v1"
 	storagev1beta1 "k8s.io/api/storage/v1beta1"
-	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/kubernetes/pkg/scheduler/algorithm"
+	"k8s.io/kubernetes/pkg/scheduler/algorithm/predicates"
```

Comment:
our cache interface should not depend on `predicates`. Other components such as `predicates` are users of `Cache`. It makes sense for them to depend on `Cache`, not the other way round. In fact, we should remove the dependency on `algorithm` that I added in the past. Please move the types you need from `predicates` to here if possible.

Reply:
It seems that nor `predicates` nor `algorithms` currently depend on Cache. And I don't think it's a good idea to make them do so. Maybe all these interfaces should go to a brand new package `pkg/scheduler/listers`. If you think that's a good idea, we should do that in a separate PR and leave this PR with the embedded interfaces.

Reply:
I know that there is no direct dependency on `cache` in `predicates` or `algorithms`, but they actually use data from cache. `NodeInfo` which is heavily used in `predicates` and `priorities` is coming from cache. In fact, we wanted to move `NodeInfo` into our `cache` directory, but since kubelet uses it, we couldn't put in under our `internal` package.
We can create a separate package for listers at `pkg/scheduler/internal/listers`. Whether you want to do it in this PR or a follow-up is up to you, but please do not create dependency in cache on `predicates`.

Reply:
Reverting. However, we still to move `NodeInfo`, without it being in an internal package.

I created #80372 to track this.